py/server/tests/test_arrow.py

```diff
@@ -0,0 +1,183 @@
+from typing import List, Any
+
+import numpy as np
+import pandas as pd
+import pyarrow as pa
+import pyarrow.parquet as papq
+
+from deephaven import arrow as dharrow, dtypes, new_table, time_table
+from deephaven.column import byte_col, char_col, short_col, int_col, long_col, float_col, double_col, \
+    string_col, datetime_col, bool_col
+from deephaven.table import Table
+from tests.testbase import BaseTestCase
+
+
+class ArrowTestCase(BaseTestCase):
+    test_table: Table
+
+    @classmethod
+    def setUpClass(cls) -> None:
+        cols = [
+            bool_col(name="Boolean", data=[True, False]),
+            byte_col(name="Byte", data=(1, -1)),
+            char_col(name="Char", data='-1'),
+            short_col(name="Short", data=[1, -1]),
+            int_col(name="Int", data=[1, -1]),
+            long_col(name="Long", data=[1, -1]),
+            long_col(name="NPLong", data=np.array([1, -1], dtype=np.int8)),
+            float_col(name="Float", data=[1.01, -1.01]),
+            double_col(name="Double", data=[1.01, -1.01]),
+            string_col(name="String", data=["foo", "bar"]),
+            datetime_col(name="Datetime", data=[dtypes.DateTime(1), dtypes.DateTime(-1)]),
+        ]
+        cls.test_table = new_table(cols=cols)
+
+    @classmethod
+    def tearDownClass(cls) -> None:
+        del cls.test_table
+
+    def verify_type_conversion(self, pa_types: List[pa.DataType], pa_data: List[Any]):
+        fields = [pa.field(f"f{i}", ty) for i, ty in enumerate(pa_types)]
+        schema = pa.schema(fields)
+        pa_table = pa.table(pa_data, schema=schema)
+        dh_table = dharrow.to_table(pa_table)
+        arrow_table = dharrow.to_arrow(dh_table)
+        self.assertEqual(dh_table.size, 2)
+        self.assertTrue(pa_table.equals(arrow_table))
+
+    def test_arrow_types_bool(self):
+        pa_types = [
+            pa.bool_(),
+        ]
+        pa_data = [
+            pa.array([True, False]),
+        ]
+        self.verify_type_conversion(pa_types=pa_types, pa_data=pa_data)
+
+    def test_arrow_types_integers(self):
+        with self.subTest("signed integers"):
+            pa_types = [
+                pa.int8(),
+                pa.int16(),
+                pa.int32(),
+                pa.int64(),
+            ]
+            pa_data = [
+                pa.array([2 ** 7 - 1, -2 ** 7 + 1]),
+                pa.array([2 ** 15 - 1, -2 ** 15 + 1]),
+                pa.array([2 ** 31 - 1, -2 ** 31 + 1]),
+                pa.array([2 ** 63 - 1, -2 ** 63 + 1]),
+            ]
+            self.verify_type_conversion(pa_types=pa_types, pa_data=pa_data)
+
+    @unittest.skip("Not correctly widened")
+    def test_arrow_types_unsigned_integers(self):
+        with self.subTest("unsigned integers"):
+            pa_types = [
+                pa.uint16(),
+            ]
+            pa_data = [
+                pa.array([2 ** 16 - 1, 0]),
+            ]
+
+    @unittest.skip("Not correctly converted by DH")
```

Comment:
This looks like a problem.  We have a number of supported data types that have disabled unit tests and apparently do not work.

Reply:
I appreciate having tests that I can enable once the java side supports new (or more-correct) conversions. Let's leave them in.